doc/decisions/man_pages.md

```diff
@@ -0,0 +1,76 @@
+# Man Pages
+
+## Problem
+
+Our man pages are written as Markdown in doc/help and then converted to roff and stored in doc/man.
+This was a workaround, because `ronn-ng` isn't available on most distributions and we want to avoid packages without man pages are being built.
+Storing generated files is annoying, as it requires:
+
+- developers to always update generated files if the sources are changed
+- developers not committing irrelevant changes to generated files (e.g. as may occur with different CMAKE_INSTALL_PREFIX etc.)
+- require extra effort for continuous integration, e.g. [#4542](https://issues.libelektra.org/4542)
+
+## Constraints
+
+1. we want beautiful rendered man pages, e.g., OPTIONS section looks like normal man pages, see in Notes¹ below
+2. we cannot require rare tools for the build process: the man pages must be present in every package
+
+## Assumptions
+
+## Considered Alternatives
+
+0. `ronn-ng`, which doesn't have packages on most distribution (violates constraint 2.) and thus created this problem
+1. Write a tool that converts our specification, similar to [pythongen](/src/tools/pythongen/template/template.man)
+2. Write a tool that parses gopts `--help` output
+3. [help2man](https://www.gnu.org/software/help2man/)
+4. Doxygen:
+   - Constraint 1 probably broken
+5. Pandoc:
+   - has a few standard dependencies
+   - would need rewrite of the current documentation in doc/help:
+     - To fulfill Constraint 1 [definition lists](https://pandoc.org/MANUAL.html#definition-lists) would be needed
```

Comment:
Do we really need to rewrite the documentation, or would a "simple" `awk`/`sed` script be enough to fulfill this?

Reply:
The rewrite process can definitely be automated. It's a one-time process, so we don't even need to use standard POSIX tools.

Reply:
> It's a one-time process

In theory I agree, however we need to make sure that the updated markdown files with definition-lists syntax will also render correctly in at least:
- GitHub
- The Libelektra Website

Reply:
AFAICT Github doesn't support definition lists. But I don't think that's a big deal. Few people will try to read the manpages on Github and except a perfect rendering. We could also use a different file extension instead of `.md` (e.g. `.pandoc`) then GitHub would just show the source and not try to render it.

For the website, we can simply use `pandoc` to generate HTML and use that.